server/tree.js

```diff
@@ -0,0 +1,46 @@
+const path = require('path');
+const fs = require('fs');
+
+function isFile(content) {
+  return new Promise((resolve, reject) => {
+    fs.stat(content, (err, stats) => {
+      if (err) {
+        reject(err);
+      }
+
+      resolve(stats.isFile());
+    });
+  });
+}
+
+function makeAnswer(directoryContent, pathParam) {
+  return Promise.all(directoryContent.map(content => {
+    return isFile(path.resolve(pathParam, content))
+      .then(ifFile => {
+        return {
+          type: ifFile ? 'file' : 'directory',
+          name: content
+        };
+      });
+  }));
+}
+
+function getDirectoryContent(pathParam) {
+  return new Promise((resolve, reject) => {
+    if (!pathParam || !pathParam.trim().length) {
```

Comment:
What you think, if I will check this case in app.get? All checks of ?path= will be in one place.